cpp/tools/parquet/parquet_dump_footer.cc

```diff
@@ -0,0 +1,245 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <cstdint>
+#include <cstring>
+#include <exception>
+#include <fstream>
+#include <iostream>
+#include <limits>
+#include <optional>
+#include <random>
+
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+
+#include "arrow/filesystem/filesystem.h"
+#include "parquet/thrift_internal.h"
+#include "generated/parquet_types.h"
+
+using apache::thrift::protocol::TCompactProtocol;
+using apache::thrift::transport::TMemoryBuffer;
+using apache::thrift::transport::TTransport;
+
+namespace {
+int PrintHelp() {
+  std::cerr << R"(
+Usage: parquet-dump-footer
+  -h|--help    Print help and exit
+  --no-scrub   Do not scrub potentially PII metadata
+  --json       Output JSON instead of binary
+  --in         Input file: required
+  --out        Output file: defaults to stdout
+
+  Dumps the footer of a Parquet file to stdout or a file, optionally with
+  potentially PII metadata scrubbed.
+)";
+  return 1;
+}
+
+uint32_t ReadLE32(const void* p) {
+  auto* b = reinterpret_cast<const uint8_t*>(p);
+  return uint32_t{b[3]} << 24 | uint32_t{b[2]} << 16 | uint32_t{b[1]} << 8 |
+         uint32_t{b[0]};
+}
+
+void AppendLE32(uint32_t v, std::string* out) {
+  out->push_back(v & 0xff);
+  out->push_back((v >> 8) & 0xff);
+  out->push_back((v >> 16) & 0xff);
+  out->push_back((v >> 24) & 0xff);
+}
+
+template <typename T>
+bool Deserialize(const char* data, uint32_t len, T* obj) {
+  parquet::ThriftDeserializer des(10 << 20, 10 << 20);
```

Comment:
Should we denote the field meaning like 
```suggestion
  parquet::ThriftDeserializer des(/*string_size_limit=*/10 << 20, /*container_size_limit=*/10 << 20);
```

Reply:
Done.